components/org.wso2.carbon.identity.recovery/src/test/java/org/wso2/carbon/identity/recovery/password/NotificationPasswordRecoveryManagerTest.java

```diff
@@ -254,4 +268,61 @@ public void testSendRecoveryNotificationEmailTemplate(boolean isEMailOtpEnabled,
         jdbcRecoveryDataStoreMockedStatic.close();
 
     }
+
+    @Test
+    public void testUpdateNewPassword_checkPasswordValidityIsTriggered() throws Exception {
+
+        NotificationPasswordRecoveryManager manager = NotificationPasswordRecoveryManager.getInstance();
+
+        User user = new User();
+        user.setTenantDomain("carbon.super");
+
+        UserRecoveryData recoveryData = mock(UserRecoveryData.class);
+        when(recoveryData.getUser()).thenReturn(user);
+
+        try (MockedStatic<IdentityTenantUtil> tenantUtilMock = mockStatic(IdentityTenantUtil.class)) {
+            tenantUtilMock.when(() -> IdentityTenantUtil.getTenantId("carbon.super")).thenReturn(-1234);
+
+            RealmService realmService = mock(RealmService.class);
+            UserRealm userRealm = mock(UserRealm.class);
+            UserStoreManager userStoreManager = mock(UserStoreManager.class);
+
+            when(realmService.getTenantUserRealm(-1234)).thenReturn(userRealm);
+            IdentityRecoveryServiceDataHolder.getInstance().setRealmService(realmService);
+
+            when(userRealm.getUserStoreManager()).thenReturn(userStoreManager);
+            doThrow(buildUserStoreException()).when(userStoreManager).updateCredentialByAdmin(anyString(), anyString());
+
+            Method method = NotificationPasswordRecoveryManager.class.getDeclaredMethod("updateNewPassword", User.class,
+                    String.class, String.class, UserRecoveryData.class, boolean.class);
+            method.setAccessible(true);
+
+            try {
+                method.invoke(manager, user, "Password@123!", "user1", recoveryData, false);
+                fail("Expected an IdentityRecoveryClientException was not thrown");
+            } catch (InvocationTargetException ex) {
+                Throwable actual = ex.getCause();
+
+                assertTrue(actual instanceof IdentityRecoveryClientException);
+                IdentityRecoveryClientException identityRecoveryClientException =
+                        (IdentityRecoveryClientException) actual;
+
+                assertEquals(identityRecoveryClientException.getErrorCode(),
+                        IdentityRecoveryConstants.ErrorMessages.ERROR_CODE_PRE_UPDATE_PASSWORD_ACTION_FAILURE.getCode());
+                assertEquals(identityRecoveryClientException.getMessage(), EXTENSION_ERROR_MESSAGE);
+                assertEquals(identityRecoveryClientException.getDescription(), EXTENSION_ERROR_DESCRIPTION);
+            }
+        }
+    }
+
+    private UserStoreException buildUserStoreException() {
```

Comment:
This should be specific to Action execution right ? Better rename the method to denote that

Reply:
Fixed with d8a7e0f199ca0dd41ddeb24807f49e89527d6b61